ast/src/parsed/mod.rs

```diff
@@ -349,7 +349,7 @@ impl<R> Children<Expression<R>> for EnumVariant<Expression<R>> {
 #[derive(Debug, PartialEq, Eq, PartialOrd, Ord, Clone, Serialize, Deserialize, JsonSchema)]
 pub struct TraitImplementation<Expr> {
     pub name: SymbolPath,
-    pub type_scheme: TypeScheme,
+    pub type_scheme: TypeScheme<Expr>,
```

Comment:
I don't think this is correct. As with TypedExpression, we need the flexibility for `functions` and `type_scheme` to use different type var values, because they are used differently. For type, it is u64 after analysis, for expressions, it is analyzed::Expression.

Maybe we should talk about this in person, it's a bit tricky.